CONTRIBUTION.md

```diff
@@ -0,0 +1,223 @@
+Contributing changes
+====================
+
+Licensing of contributed material
+---------------------------------
+
+Keep in mind as you contribute, that code, docs and other material submitted to
+open source projects are usually considered licensed under the same terms
+as the rest of the work.
+
+The details vary from project to project, but from the perspective of this
+document's authors:
+
+- Anything submitted to a project falls under the licensing terms in the
+  repository's top level ``LICENSE`` file.
+
+    - For example, if a project's ``LICENSE`` is BSD-based, contributors should
+      be comfortable with their work potentially being distributed in binary
+      form without the original source code.
+
+- Per-file copyright/license headers are typically extraneous and undesirable.
+  Please don't add your own copyright headers to new files unless the project's
+  license actually requires them!
+
+    - Not least because even a new file created by one individual (who often
+      feels compelled to put their personal copyright notice at the top) will
+      inherently end up contributed to by dozens of others over time, making a
+      per-file header outdated/misleading.
+
+Version control branching
+-------------------------
+
+* Always **fork the repo** or **make a new branch** for your work, no matter how small. This makes
+  it easy for others to take just that one set of changes from your repository,
+  in case you have multiple unrelated changes floating around.
+
+    * A corollary: **don't submit unrelated changes in the same branch/pull
+      request**! The maintainer shouldn't have to reject your awesome bugfix
+      because the feature you put in with it needs more review.
+
+* **Base your new branch off of the appropriate branch** on the main
+  repository:
+
+    * **Bug fixes** should be based on the branch named after the **oldest
+      supported release line** the bug affects.
+
+        * E.g. if a feature was introduced in 1.1, the latest release line is
+          1.3, and a bug is found in that feature - make your branch based on
+          1.1.  The maintainer will then forward-port it to 1.3 and master.
+        * Bug fixes requiring large changes to the code or which have a chance
+          of being otherwise disruptive, may need to base off of **master**
+          instead. This is a judgement call -- ask the devs!
+
+    * **New features** should branch off of **the 'master' branch**.
+
+        * Note that depending on how long it takes for the dev team to merge
+          your patch, the copy of ``master`` you worked off of may get out of
+          date! If you find yourself 'bumping' a pull request that's been
+          sidelined for a while, **make sure you rebase or merge to latest
+          master** to ensure a speedier resolution.
```

Comment:
```suggestion
Version control branching
-------------------------

There are 2 active branches on this repo, plus branches for each major version (see below):
- The `master` branch is the main branch of the repository, the one that is used for active development
  - Every time this branch is modified (typically when merging a PR), a Storybook is automatically deployed (on _Radix_, `master` environment). It can be found [here](TODO: add link)
  - Every time this branch is modified, an updated package is deployed to [npm](TODO: add link)
- The `latest` branch always points at the the most recent version released
  - The branch has to be updated manually, whenever a new release is created
  - Every time this branch is updated, the Storybook is automatically deployed (on _Radix_, `latest` environment). It can be found [here](TODO: add link)

This repository relies on [Semantic Versioning](https://semver.org/).
 
In order to create a release for a new major version, the first step is to create a new branch named `version_X` in the repo (where `X` has to be replaced with an incremental number). Then, a new _Release_ is created from that branch here on Github, named `vX.0.0`. Remember to update the `latest` branch to the new release.

Development continues on the `master` branch. In order to create a new minor version, the desired changes have to be moved to the `version_X` branch (either by merging, rebasing, cherry-picking, …). After proper testing, a new _Release_ can be created here on Github, named `vX.Y.0`. Remember to update the `latest` branch to the new release.

If a hotfix has to be applied to a released version, the fix has to be added on top of `version_X`. Then a new _Release_ can be created here on Github, named `vX.Y.Z`. Remember to update the `latest` branch to the new release. If the fix is relevant also for `master`, make sure the fix is ported on top of `master` as well.

When you want to contribute to the repo, always **fork the repo** or **make a new branch** for your work, no matter how small. This makes it easy for others to take just that one set of changes from your repository, in case you have multiple unrelated changes floating around.

    * A corollary: **don't submit unrelated changes in the same branch/pull request**!

* **Base your new branch off of the appropriate branch** on the main repository:

    * **Bug fixes** should be based either on `master` or on `latest` branches. Contribution to an older `version_X` branch (i.e., any version branch that does not correspond with `latest`) will be **rejected** (there can be exceptions, but that would require very compelling arguments). 

        * Bug fixes requiring large changes to the code or which have a chance
          of being otherwise disruptive, may need to based on top of **master**.
          This is a judgement call -- ask the devs!

    * **New features** should branch off of the `master` branch

        * Note that depending on how long it takes for the dev team to merge
          your patch, the copy of ``master`` you worked off of may get out of
          date! If you find yourself 'bumping' a pull request that's been
          sidelined for a while, **make sure you rebase or merge to latest
          master** to ensure a speedier resolution.
```

Reply:
i think this commit suggestion would be better as a pull request into this branch, I would like to see some of it changed: "  - Every time this branch is modified, an updated package is deployed to [npm](TODO: add link)", we don't have a fully automatic deployment to npm, i'm also not sure if we should.

We should also describe our deployment process, that we need to bump the version in  `package-lock` and `package.json`. That we create a release in GitHub that triggers an action that pushes to npm. And that the newest stuff in the changelog would be a part of the details for the release

Reply:
Sounds reasonable. Let's merge this then, and we will consider further changes in dedicated PRs